docs/testing/unit-testing.md

```diff
@@ -0,0 +1,413 @@
+- Because the Office JavaScript APIs library isn't loaded in the node process, the `Word` object that is referenced in the add-in code must be declared and initialized.
+
+```javascript
+const OfficeAddinMock = require("office-addin-mock");
+const myWordAddinFeature = require("../my-word-add-in-feature");
+
+// Create the seed mock object.
+const mockData = {
+  context: {
+    document: {
+      body: {
+        paragraph: {
+          font: {},
+        },
+        // Mock the Body.insertParagraph method.
+        insertParagraph: function (paragraphText, insertLocation) {
+          this.paragraph.text = paragraphText;
+          this.paragraph.insertLocation = insertLocation;
+          return this.paragraph;
+        },
+      },
+    },
+  },
+  // Mock the Word.InsertLocation enum.
+  InsertLocation: {
+    end: "end",
+  },
+  // Mock the Word.run method.
+  run: async function(callback) {
+    await callback(this.context);
+  },
+};
+
+// Create the final mock object from the seed object.
+const wordMock = new OfficeAddinMock.OfficeMockObject(mockData);
+
+// Define and initialize the Word object that is called in the insertBlueParagraph function.
+global.Word = wordMock;
+
+/* Code that calls the test framework goes below this line. */
+
+// Jest test set
+describe("Insert blue paragraph at end tests", () => {
+
+  test("color of paragraph", async function () {
+    await myWordAddinFeature.insertBlueParagraph();  
+    expect(wordMock.context.document.body.paragraph.font.color).toBe("blue");
+  });
+
+  test("text of paragraph", async function () {
+    await myWordAddinFeature.insertBlueParagraph();
+    expect(wordMock.context.document.body.paragraph.text).toBe("Hello World");
+  });
+})
+```
+
+## Adding mock objects, properties, and methods dynamically when testing
+
+In some scenarios, efficient testing requires that mock objects be created or modified at runtime; that is, while the tests are running. The following are examples:
+
+- The function that is being tested behaves differently when called a second time, from how it behaves when first called. You need to first test the function with one mock object, then change this mock object and test the function again with the changed mock object.
+- You need to test a function against multiple arguments quickly. For example, you need to test a function with a mock object that has a color property and then test the function again with a mock object that has a text property, but is otherwise identical to the original mock object.
+
+The `OfficeMockObject` has three methods to help in these scenarios.
+
+- `OfficeMockObject.setMock` adds a property and value to a `OfficeMockObject` object. The following is an example:
+
+    ```javascript
+    rangeMock.setMock("address", "G6:K9");
+    ```
+
+- `OfficeMockObject.addMockFunction` adds a mock function to a `OfficeMockObject` object. If the function parameter isn't used, an empty function is created. The following is an example:
```

Comment:
It's a little confusing to read about the empty parameter case before seeing an example that doesn't use it. Maybe pull that line out as a note that follows the sample code?